From linas@austin.ibm.com  Mon Nov 26 15:10:36 2007
From: Linas Vepstas <linas@austin.ibm.com>
Date: Fri, 2 Nov 2007 15:14:28 -0500
Subject: PCI: pcie portdriver: initialize returned value
To: tom.l.nguyen@intel.com, gregkh@suse.de
Cc: linux-pci@atrey.karlin.mff.cuni.cz
Message-ID: <20071102201428.GT2183@austin.ibm.com>
Content-Disposition: inline

From: Linas Vepstas <linas@austin.ibm.com>

The pcie protdrv status can be returned uninitialized,
if there are no children under a device. This leads to
bad responses downstream. Fix this.

Signed-off-by: Linas Vepstas <linas@austin.ibm.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/pcie/portdrv_pci.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/pci/pcie/portdrv_pci.c
+++ b/drivers/pci/pcie/portdrv_pci.c
@@ -217,7 +217,7 @@ static int slot_reset_iter(struct device
 
 static pci_ers_result_t pcie_portdrv_slot_reset(struct pci_dev *dev)
 {
-	pci_ers_result_t status;
+	pci_ers_result_t status = PCI_ERS_RESULT_NONE;
 	int retval;
 
 	/* If fatal, restore cfg space for possible link reset at upstream */
